### PR TITLE
lambda: make it runs in go1.x rather than python2.7

### PR DESCRIPTION
**Pull request by xlr-8**
# Issue Type

- Enhancement Pull Request

## Summary

The code was currently running as a binary called via python2.7,
however since the annoucement of go support for lambda by AWS, it makes
more sense to conf/push it directly into that environment and benefit
from both the performance, support and monitoring/tracing.

Closes: https://github.com/cristim/autospotting/issues/170

## Code contribution checklist

1. [x] The contribution fixes a single existing github issue, and it is linked
   to it.
1. [x] The code is as simple as possible, readable and follows the idiomatic Go
  [guidelines](https://golang.org/doc/effective_go.html).
1. [x] All new functionality is covered by automated test cases so the overall
  test coverage doesn't decrease.
1. [x] No issues are reported when running `make full-test`.
1. [x] Functionality not applicable to all users should be configurable.
1. [x] Configurations should be exposed through Lambda function environment
   variables which are also passed as parameters to the
   [CloudFormation](https://github.com/cristim/autospotting/blob/master/cloudformation/stacks/AutoSpotting/template.json)
   and
   [Terraform](https://github.com/cristim/autospotting/blob/master/terraform/autospotting.tf)
   stacks defined as infrastructure code.
1. [x] Global configurations set from the infrastructure stack level should also
   support per-group overrides using tags.
1. [x] Tags names and expected values should be similar to the other existing
   configurations.
1. [x] Both global and tag-based configuration mechanisms should be tested and
  proven to work using log output from various test runs.
1. [x] The logs should be kept as clean as possible (use log levels as
  appropriate) and formatted consistently to the existing log output.
1. [x] The documentation is updated to cover the new behavior, as well as the
   new configuration options for both stack parameters and tag overrides.
1. [x] A code reviewer reproduced the problem and can confirm the code
  contribution actually resolves it.

### CODE REVIEW

File: autospotting.go [hunk 1]
```diff
@@ -26,7 +26,7 @@ var Version string
 var ExpirationDate string
 
 func main() {
-	run()
+	lambda.Start(Handler)
```
Comment (cristim) [inline autospotting.go]: I think we can use one of the [reserved environment variables](
https://docs.aws.amazon.com/lambda/latest/dg/current-supported-versions.html) to determine if the binary is executed in the Lambda environment so we then run the handler, and keep calling the `run()` when executed locally, like we do now. A simple if like this one should do the trick (note: this is just a hunch, and it wasn't yet tested): 

```
if os.Getenv("AWS_LAMBDA_FUNCTION_NAME") != "" {
  lambda.Start(Handler)
} else {
 run()
}
```

Comment (xlr-8) [reply autospotting.go]: Exactly my thought

**Comment by xlr-8** [comment]
The small fix for checking environment seem to work just fine, I haven't tested cloudformation yet though.
I'd like someone to double check it as well if possible.

**Comment by cristim** [comment]
I tested it with CloudFormation and it worked like a charm once i updated my stack configuration.

**Pull request merged**